packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -59,24 +59,49 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
     protected readonly onWillSaveModelEmitter = new Emitter<WillSaveMonacoModelEvent>();
     readonly onWillSaveModel = this.onWillSaveModelEmitter.event;
 
+    private preferredEncoding: string | undefined = undefined;
+    private readonly defaultEncoding: string | undefined;
+
     constructor(
         protected readonly resource: Resource,
         protected readonly m2p: MonacoToProtocolConverter,
-        protected readonly p2m: ProtocolToMonacoConverter
+        protected readonly p2m: ProtocolToMonacoConverter,
+        options?: { encoding?: string | undefined }
     ) {
         this.toDispose.push(resource);
         this.toDispose.push(this.toDisposeOnAutoSave);
         this.toDispose.push(this.onDidChangeContentEmitter);
         this.toDispose.push(this.onDidSaveModelEmitter);
         this.toDispose.push(this.onWillSaveModelEmitter);
         this.toDispose.push(this.onDirtyChangedEmitter);
-        this.resolveModel = resource.readContents().then(content => this.initialize(content));
+        this.resolveModel = resource.readContents(options).then(content => this.initialize(content));
+        this.defaultEncoding = options && options.encoding ? options.encoding : undefined;
     }
 
     dispose(): void {
         this.toDispose.dispose();
     }
 
+    async reopenWithEncoding(encoding: string) {
```

Comment:
please rebase and apply latest tslint rules, `return types` are mandator according to the coding guidelines

Reply:
Done